runtime/src/main/java/io/micronaut/scheduling/executor/ExecutorFactory.java

```diff
@@ -53,31 +69,30 @@ public ExecutorService executorService(ExecutorConfiguration executorConfigurati
         ExecutorType executorType = executorConfiguration.getType();
         switch (executorType) {
             case FIXED:
-                return executorConfiguration
-                    .getThreadFactoryClass()
-                    .flatMap(InstantiationUtils::tryInstantiate)
-                    .map(factory -> Executors.newFixedThreadPool(executorConfiguration.getNumberOfThreads(), factory))
-                    .orElse(Executors.newFixedThreadPool(executorConfiguration.getNumberOfThreads(), threadFactory));
-
+                return Executors.newFixedThreadPool(executorConfiguration.getNumberOfThreads(), getThreadFactory(executorConfiguration));
             case CACHED:
-                return executorConfiguration
-                    .getThreadFactoryClass()
-                    .flatMap(InstantiationUtils::tryInstantiate)
-                    .map(Executors::newCachedThreadPool)
-                    .orElse(Executors.newCachedThreadPool(threadFactory));
-
+                return Executors.newCachedThreadPool(getThreadFactory(executorConfiguration));
             case SCHEDULED:
-                return executorConfiguration
-                    .getThreadFactoryClass()
-                    .flatMap(InstantiationUtils::tryInstantiate)
-                    .map(factory -> Executors.newScheduledThreadPool(executorConfiguration.getCorePoolSize(), factory))
-                    .orElse(Executors.newScheduledThreadPool(executorConfiguration.getCorePoolSize(), threadFactory));
-
+                return Executors.newScheduledThreadPool(executorConfiguration.getCorePoolSize(), getThreadFactory(executorConfiguration));
```

Comment:
I believe we can just add `@Nullable @Parameter ThreadFactory threadFactory` to avoid having to do the manually bean locator lookups

Reply:
SCHEDULED doesn't need a `threadFactory`